android/src/main/java/nordpol/TagDispatcher.java

```diff
@@ -10,68 +10,76 @@
-        this.nfcDisabledListener = nfcDisabledListener;
+        this.handleUnavailableNfc = handleUnavailableNfc;
     }
 
     public static TagDispatcher get(Activity activity,
                                     OnDiscoveredTagListener tagDiscoveredListener,
-                                    OnNfcDisabledListener nfcDisabledListener) {
-        return new TagDispatcher(activity, tagDiscoveredListener, nfcDisabledListener);
+                                    boolean handleUnavailableNfc) {
+        return new TagDispatcher(activity, tagDiscoveredListener, handleUnavailableNfc);
     }
 
     public static TagDispatcher get(Activity activity,
                                     OnDiscoveredTagListener tagDiscoveredListener) {
-        return new TagDispatcher(activity, tagDiscoveredListener, null);
+        return new TagDispatcher(activity, tagDiscoveredListener, true);
     }
 
 
     /** Enable exclusive NFC access for the given activity.
      * Using this method makes NFC intent filters in the AndroidManifest.xml redundant.
-     * @returns true if NFC was available and false if no NFC is available
-     *          on device.
+     * @returns NfcStatus.AVAILABLE_ENABLED if NFC was available and enabled,
+     * NfcStatus.AVAILABLE_DISABLED if NFC was available and disabled and
+     * NfcStatus.NOT_AVAILABLE if no NFC is available on the device.
      */
     @TargetApi(Build.VERSION_CODES.KITKAT)
-    public boolean enableExclusiveNfc() {
+    public NfcStatus enableExclusiveNfc() {
         NfcAdapter adapter = NfcAdapter.getDefaultAdapter(activity);
         if (adapter != null) {
             if (!adapter.isEnabled()) {
-                if (null == nfcDisabledListener) {
+                if (handleUnavailableNfc) {
                     activity.startActivity(new Intent(
                     android.provider.Settings.ACTION_NFC_SETTINGS));
-                } else {
-                    nfcDisabledListener.nfcDisabled();
                 }
-                return false;
+                return NfcStatus.AVAILABLE_DISABLED;
             }
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
                 enableReaderMode(adapter);
             } else {
                 enableForegroundDispatch(adapter);
             }
-            return true;
+            return NfcStatus.AVAILABLE_ENABLED;
         }
-        return false;
+        if (handleUnavailableNfc) toast("NFC is not available on this device");
+        return NfcStatus.NOT_AVAILABLE;
     }
 
     /** Disable exclusive NFC access for the given activity.
-     * @returns true if NFC was available and false if no NFC is available
-     *          on device.
+     * @returns NfcStatus.AVAILABLE if NFC was available and
+     * NfcStatus.NOT_AVAILABLE if no NFC is available on the device.
      */
     @TargetApi(Build.VERSION_CODES.KITKAT)
-    public boolean disableExclusiveNfc() {
+    public NfcStatus disableExclusiveNfc() {
```

Comment:
@Franzaine Maybe this doesn't really makes sense. I guess we could just as well make this function void?